File: MediCareWebApi/Models/Prescription.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class Prescription
	{
		[Required]
		public DateTime IssueDate { get; set; }

		[Required]
		public DateTime ExpirationDate { get; set; }

		[Required]
		[MaxLength(256)]
		public string Description { get; set; }

		[Key]
		[Required]
		public int AppointmentId { get; set; }

		[ForeignKey("AppointmentId")]
		public Appointment Appointment { get; set; }

		public ICollection<PrescriptionMedicament> PrescriptionMedicaments { get; set; } = new List<PrescriptionMedicament>();
	}
}

File: MediCareWebSpa/src/app/DTOs/models/prescription.ts
import { PrescriptionMedicament } from "./prescription-medicament";

export class Prescription {
  id: number;
  description: string;
  issueDate: Date;
  expirationDate: Date;
  appointmentId: number;
  prescriptionMedicaments: PrescriptionMedicament[] = [];

  constructor(
    id: number,
    description: string,
    issueDate: Date,
    expirationDate: Date,
    appointmentId: number
  ) {
    this.id = id;
    this.description = description;
    this.issueDate = issueDate;
    this.expirationDate = expirationDate;
    this.appointmentId = appointmentId;
  }
}
